src/App.jsx

```diff
@@ -1,247 +1,118 @@
-            </p>
-          </div>
-
-          <div className="panel-block is-flex-wrap-wrap">
-            <a
-              href="#/"
-              data-cy="AllCategories"
-              className="button is-success mr-6 is-outlined"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1 is-info"
-              href="#/"
-            >
-              Category 1
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 2
-            </a>
-
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1 is-info"
-              href="#/"
-            >
-              Category 3
-            </a>
-            <a
-              data-cy="Category"
-              className="button mr-2 my-1"
-              href="#/"
-            >
-              Category 4
-            </a>
-          </div>
-
-          <div className="panel-block">
-            <a
-              data-cy="ResetAllButton"
-              href="#/"
-              className="button is-link is-outlined is-fullwidth"
-            >
-              Reset all filters
-            </a>
-          </div>
-        </nav>
-      </div>
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+import { Filters } from './components/Filters';
+import { ProductList } from './components/ProductList';
+
+const products = productsFromServer.map((product) => {
+  const category = categoriesFromServer
+    .find(({ id }) => product.categoryId === id);
+  const user = usersFromServer
+    .find(({ id }) => category.ownerId === id);
+
+  return {
+    ...product,
+    category,
+    user,
+  };
+});
+
+function filterProducts(productsToFilter, queries) {
+  const {
+    userId,
+    searchQuery,
+    selectedCategories,
+  } = queries;
+
+  let filteredData = filterByUser(productsToFilter, userId);
+
+  filteredData = filterBySearchQuery(filteredData, searchQuery);
+  filteredData = filterByCategories(filteredData, selectedCategories);
+
+  return filteredData;
+}
+
+function filterByUser(productsToFilter, userId) {
+  if (!userId) {
+    return productsToFilter;
+  }
+
+  return productsToFilter.filter(({ user }) => user.id === userId);
+}
+
+function filterBySearchQuery(productsToFilter, searchQuery) {
+  const lowerQuery = searchQuery.toLowerCase();
+
+  return productsToFilter
+    .filter(({ name }) => name.toLowerCase().includes(lowerQuery));
+}
+
+function filterByCategories(productsToFilter, categories) {
+  if (!categories.length) {
+    return productsToFilter;
+  }
+
+  return productsToFilter
+    .filter(({ category }) => categories.includes(category.id));
+}
+
+export const App = () => {
+  const [selectedUser, setSelectedUser] = useState(0);
```

Comment:
it's not clear here why user can contain value
will be better to name it as `selectedUserId`